extensions/security/runtime/src/main/java/io/quarkus/security/runtime/interceptor/SecurityHandler.java

```diff
@@ -41,7 +41,9 @@ public Object handle(InvocationContext ic) throws Exception {
                     }).subscribeAsCompletionStage();
         } else if (Multi.class.isAssignableFrom(returnType)) {
             return constrainer.nonBlockingCheck(ic.getMethod(), ic.getParameters())
-                    .onItem().transformToMulti(new MultiContinuation(ic));
+                    // Do not use transformToMulti as we want to preserve the type of multi produced by the user method.
+                    // This is required for Reactive Routes to handle custom Multi serialization.
+                    .toMulti().plug(x -> new MultiContinuation(ic).apply(x));
```

Comment:
what's the actual difference?

Reply:
IIUIC you don't wrap the `Multi` produced by the user method but instead call `new MultiContinuation(ic).apply(x)` when the `Multi` instance is created.